compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -53,19 +41,19 @@ impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for Owner<'tcx> {
 /// The parent is trash when the node is a HIR owner.
 #[derive(Clone, Debug)]
 pub struct ParentedNode<'tcx> {
-    parent: ItemLocalId,
-    node: Node<'tcx>,
+    pub parent: ItemLocalId,
+    pub node: Node<'tcx>,
 }
 
 #[derive(Debug)]
 pub struct OwnerNodes<'tcx> {
     /// Pre-computed hash of the full HIR.
-    hash: Fingerprint,
+    pub hash: Fingerprint,
     /// Full HIR for the current owner.
     // The zeroth node's parent is trash, but is never accessed.
```

Comment:
What does "trash" mean? Is it `None`? Should it be `None`?

Reply:
Trash in the sense it is set to `ItemLocalId::new(0)`, which is wrong since it's not even inside the same owner, but it's not a big deal since client code does not access `nodes[0].parent`.